### PR TITLE
Add compilation defaults for the Fnet MaskedLM task model

### LINKED ISSUE #830: Add compilation defaults for the MaskedLM task models
In https://github.com/keras-team/keras-nlp/issues/709 we added compilation defaults to all classification tasks. We should also do the same for all `XXMaskedLM` models. E.g. [bert](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/bert/bert_masked_lm.py). Here's the full list of models.

- [ ] `AlbertMaskedLM`
- [ ] `BertMaskedLM`
- [ ] `DebertV3MaskedLM`
- [ ] `DistilBertMaskedLM`
- [ ] `FNetMaskedLM`
- [ ] `RobertaMaskedLM`

https://github.com/keras-team/keras-nlp/pull/714 is a good template PR. We can probably just use Adam and `1e-5` or `2e-5` as a learning rate for now. Though a little experimentation for each model to make sure the task does converge would be helpful. This [colab](https://gist.github.com/mattdangerw/b16c257973762a0b4ab9a34f6a932cc1) may be a helpful starting place.

### PR DESCRIPTION
**Pull request by soma2000-lang**
Partially fixes #830

### CODE REVIEW

**Comment by mattdangerw** [comment]
Code looks good. Have we tried this out with a few different presets to make sure we are converging generally?

File: keras_nlp/models/f_net/f_net_masked_lm.py [hunk 1]
```diff
@@ -135,6 +136,12 @@ def __init__(
         # All references to `self` below this line
         self.backbone = backbone
         self.preprocessor = preprocessor
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(2e-5),
+            metrics=keras.metrics.SparseCategoricalAccuracy(),
```
Comment (mattdangerw) [inline keras_nlp/models/f_net/f_net_masked_lm.py:142]: Actually, I think we want weighted_metrics here as we do pass sample weights for this task.

**Comment by chenmoneygithub** [comment]
Thanks for the PR!  🥂

@mattdangerw Matt, I am feeling for future PRs adding compilation defaults, we may want to require fitting results to justify the value we choose. It will add a bit overhead to such PRs, but I think it's important for us to ensure the correctness.

**Comment by mattdangerw** [comment]
> @mattdangerw Matt, I am feeling for future PRs adding compilation defaults, we may want to require fitting results to justify the value we choose.

Agreed! I think this can be kinda add hoc for now, as we didn't really have an established process for classification defaults either. But long term, we for sure should invest in infra to experiment with multiple learning rates for multiple checkpoints (and potentially even multiple datasets!), to make sure we have good default. Obviously we are limited by compute and will have to make some tradeoffs.

For now, let's just sanity check that our constant are somewhat reasonable.

**Comment by chenmoneygithub** [comment]
@mattdangerw yea, budget for computing is a pain for us now. I think for now we can ask for sanity check on small/tiny from users, since they can fit in colab free-tier GPU. But as we don't have a guideline right now, we can merge this PR without asking for that.

**Comment by chenmoneygithub** [comment]
@mattdangerw I opened an [issue](https://github.com/keras-team/keras-nlp/issues/850) to track.

**Comment by shivance** [comment]
> @mattdangerw yea, budget for computing is a pain for us now. I think for now we can ask for sanity check on small/tiny from users, since they can fit in colab free-tier GPU. But as we don't have a guideline right now, we can merge this PR without asking for that.

Hi @chenmoneygithub, I think we can consider using Kaggle Notebooks as an alternative to colab free tier ( colab gpus are pretty slow compared to kaggle , see [this](https://twitter.com/anshum0n/status/1633548651787333633?s=20). For #833 , following your example script, I am experimenting with 1e-5, 2e-5, 5e-5, 1e-4 as LRs for the three models. 

[Script](https://www.kaggle.com/code/shivanshuman/does-tensorflow-task-converge) is still running, while it demonstrates how to brute-force check with which LR model will perform better. Do you think it'll be a meaningful contribution against #850 ?

On a side note, I don't remember if colab still has that 1 hr limit or so, but kaggle doesn't. 

image

I think we can use kaggle (private notebook if needed) to  do tpu testing for new backbones we add.

Thanks.

**Comment by chenmoneygithub** [comment]
@shivance It's good to know! thanks!

File: keras_nlp/models/f_net/f_net_masked_lm.py (hunk 1 above)
Comment (soma2000-lang) [reply keras_nlp/models/f_net/f_net_masked_lm.py:142]: Sure

Comment (soma2000-lang) [reply keras_nlp/models/f_net/f_net_masked_lm.py:142]: Still working on the converging part

**Comment by soma2000-lang** [comment]
@mattdangerw @chenmoneygithub  I have tested this performs the best at learning rate 1e-4.Accordingly updated the value.

**Comment by mattdangerw** [comment]
@soma2000-lang thanks! Do you have a way to share the experiments you did?

I think going off of @shivance testing on https://github.com/keras-team/keras-nlp/pull/833, it seems like 5e-5 might be a good general purpose learning rate here, and a lower number will be a little more conservative. Let's just stick with that for now for simplicity, but we can always update.

**Comment by soma2000-lang** [comment]
@mattdangerw 
image
